src/dbc.rs

```diff
@@ -345,8 +361,9 @@ pub(crate) mod tests {
             .build(rng)?;
 
         for (key_image, tx) in dbc_builder.inputs() {
-            dbc_builder =
-                dbc_builder.add_spent_proof_share(spentbook_node.log_spent(key_image, tx)?);
+            dbc_builder = dbc_builder
+                .add_spent_proof_share(spentbook_node.log_spent(key_image, tx.clone())?)
+                .add_spent_transaction(tx);
```

Comment:
this looks like there will be `one transaction per proof_share` get added, which means X copies of duplicated tx get pushed to the vector?

Reply:
it seems so, but with the change to using a set for the TXs then we should be fine